content/en/blog/2021/stateful-set-made-easier/index.md

```diff
@@ -0,0 +1,223 @@
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37498         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37384         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37514         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37402         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37434         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37526         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37374         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37442         TIME_WAIT
+tcp        0      0 127.0.0.1:2181          127.0.0.1:37464         TIME_WAIT
+{{< /text >}}
+
+There is nothing `ESTABLISHED` on port 2888 or 3888.  Next, let us get the ZooKeeper server status:
+
+{{< text bash plain >}}
+$ kubectl exec my-release-zookeeper-1 -c zookeeper -- /opt/bitnami/zookeeper/bin/zkServer.sh status
+/opt/bitnami/java/bin/java
+ZooKeeper JMX enabled by default
+Using config: /opt/bitnami/zookeeper/bin/../conf/zoo.cfg
+Client port found: 2181. Client address: localhost. Client SSL: false.
+Error contacting service. It is probably not running.
+{{< /text >}}
+
+From the above output, you can see the ZooKeeper service is not functioning properly. Let us check the cluster configuration for one of the ZooKeeper pods:
+
+{{< text bash plain >}}
+$ istioctl proxy-config cluster my-release-zookeeper-1 --port 3888 --direction inbound -o json
+[
+    {
+        "name": "inbound|3888||",
+        "type": "STATIC",
+        "connectTimeout": "10s",
+        "loadAssignment": {
+            "clusterName": "inbound|3888||",
+            "endpoints": [
+                {
+                    "lbEndpoints": [
+                        {
+                            "endpoint": {
+                                "address": {
+                                    "socketAddress": {
+                                        "address": "127.0.0.1",
+                                        "portValue": 3888
+                                    }
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        },
+...
+{{< /text >}}
+
+What is interesting here is that the inbound on port 3888 has 127.0.0.1 as its endpoint. This means that Envoy proxy before Istio 1.10 redirects the inbound traffic to the `loopback` interface, as described in the [upcoming networking change in 1.10 blog](/blog/2021/upcoming-networking-changes/).
+
+## StatefulSets in Action with Istio 1.10
+
+Install Istio 1.10 and configure the `default` namespace to enable Istio 1.10’s sidecar injection. Let’s rolling restart the ZooKeeper StatefulSet to update the pods to use Istio 1.10’s sidecar proxy:
+
+{{< text bash >}}
+$ kubectl rollout restart statefulset my-release-zookeeper
+{{< /text >}}
```

Comment:
Would it be more powerful if you installed again from the top?

(In fact, if that's the case, maybe you should lead with this section?)

Reply:
My thoughts is to meet users where they are today given these are very common issues with stateful sets and proved that we are better while explaining how the network behavior changed.